### ISSUE TITLE
Project Licensing Changes (introduce CC0 for templates)

### ISSUE THREAD

**Issue by ForNeVeR** [ForNeVeR/Cesium#682]
(I swear no big or bad changes, please bear with me.)

**TL;DR:** I propose to migrate CC0 for project templates; everything else stays under MIT (we know of examples when this is ok for the C library).

## Long Version

I'd like to announce/discuss some small changes in how we license the code, mostly for the users' convenience.

First of all, I was unsure about licensing the standard library headers and runtime under MIT (whether the user code will then have to deal with our license and reproduce our copyrights or not), but I see how musl does the same (they license the whole musl with all the headers under MIT), so this is okay.

Then, there are also the project templates. While there are some .NET project templates that are under MIT (xUnit) and nobody actually cares and you can even argue that the project template content is not copyrightable, I prefer us to not even have this discussion with the users, and them not have this doubt at all.

So, while everything else in the compiler, SDK and any other artifacts we produce will employ [the REUSE spec](https://reuse.software/) and be properly 100% marked as MIT sooner or later (a branch is in the works already), I would like the project templates (and the content directly included in the templates exclusively) to have [CC0 license](https://creativecommons.org/public-domain/cc0/), essentially "no rights reserved" badge.

Our project templates are a handful of lines in `.ceproj` and `.c` anyway, so it's not a big deal for us — while it will be a big deal for the users who want to reuse them _properly_, attributing all the copyrights and respecting the license. By declaring these under CC0, we allow them to just freely reuse the texts and not care for a slightest about any of the stuff.

So, when the time is right™, I'll ask @seclerp to formally consent with relicensing of the template contents under CC0. Please be aware.
